### PR TITLE
Feature/add blockchain

### PR DESCRIPTION
**Pull request by mjsorribas**
## Description
Updated Pr with last Blockchain module changes.

-   adds blockchain API's
-   info
-   block
-   list
-   transaction
-   request
-   response
-  add isomorphic-fetch package

## Testing

- jest tests added

Green tests :
image

## Related Issues
Fixes https://github.com/liftedinit/many-js/issues/72

## Checklist:

- [x] I have read and followed the CONTRIBUTING guidelines for this project.
- [x] I have added or updated tests and they pass.
- [x] I have added or updated documentation and it is accurate.
- [x] I have noted any breaking changes in this module or downstream modules.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+require("isomorphic-fetch")
```
Comment (hansl) [inline src/index.ts:1]: Why is this needed? Any polyfill should be added to apps/tests harnesses using this library. Polyfills should not be required by libraries.

**Review by hansl — CHANGES_REQUESTED** [review]
See comment above. If you need it for tests, add it to the test setups, not the `src/index.ts`.

**Review by mjsorribas — COMMENTED** [review]
I removed the isomorphic-fetch, and after that, I tested Talib and Talib-be to validate if I received any errors. 
Another thing: I tried to update the     "base32-encode": "^1.2.0", to 2.0.0 but I got some errors.

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-16.16.0
+18.13.0
```
Comment (stanleyjones) [inline .nvmrc:1]: Have you tested this branch with Alberto or Gwen (which both still use Node v16)? Will it cause version conflicts with their dependencies?

File: package.json [hunk 3]
```diff
@@ -1,31 +1,30 @@
 {
   "name": "@liftedinit/many-js",
-  "version": "0.1.0-alpha.7",
+  "version": "0.1.0-alpha.8",
```
Comment (stanleyjones) [inline package.json:3]: I'd prefer to bump the version of `many-js` separately so we generate tags and GitHub releases.

File: package.json [hunk 4]
```diff
@@ -1,31 +1,30 @@
...
-    "@types/node": "^12.20.37",
-    "@types/node-forge": "^0.10.10",
-    "@types/uuid": "^8.3.3",
-    "@typescript-eslint/parser": "^5.7.0",
+    "@types/crc": "^3.8.0",
+    "@types/jest": "^29.2.6",
+    "@types/node": "^18.11.18",
+    "@types/node-forge": "^1.3.1",
+    "@types/uuid": "^9.0.0",
+    "@typescript-eslint/parser": "^5.48.0",
```
Comment (stanleyjones) [inline package.json:15]: Let's take this opportunity to move all the `@types` stuff into `devDependencies`.

File: src/network/modules/blockchain/blockchain.ts [hunk 5]
```diff
@@ -1,17 +1,355 @@
...
+  block: (
+    queryType: SingleBlockQueryType,
+    value: SingleBlockQueryValueType,
+  ) => Promise<BlockReturns>
+  list(opts: {
+    queryType: RangeBlockQueryType
+    count?: number
+    order?: ListOrderType
+    range?: RangeBounds<number> | RangeBounds<Date>
+  }): Promise<BlockListReturns>
```
Comment (stanleyjones) [inline src/network/modules/blockchain/blockchain.ts:85]: Maybe declare an interface for `opts` above instead of inline here?

File: src/network/modules/blockchain/blockchain.ts [hunk 6]
```diff
@@ -1,17 +1,355 @@
...
+  request: (txnHash: ArrayBuffer) => Promise<unknown>
+  response: (txnHash: ArrayBuffer) => Promise<unknown>
+}
+
+export const Blockchain: Blockchain = {
+  _namespace_: "blockchain",
+
+  async info(): Promise<InfoReturns> {
+    const msg = await this.call("blockchain.info")
+    return await getBlockainInfo(msg)
```
Comment (stanleyjones) [inline src/network/modules/blockchain/blockchain.ts:96]: typo: Should be `getBlockchainInfo`

File: src/network/modules/blockchain/blockchain.ts [hunk 7]
```diff
@@ -1,17 +1,355 @@
...
+    queryType,
+    count = 10,
+    order = ListOrderType.descending,
+    range = [],
+  }: {
+    queryType: RangeBlockQueryType
+    count?: number
+    order?: ListOrderType
+    range?: RangeBounds<number> | RangeBounds<Date>
+  }): Promise<BlockListReturns> {
```
Comment (stanleyjones) [inline src/network/modules/blockchain/blockchain.ts:128]: Yeah, it'd be nice to have a declared interface for list options so we don't have to repeat them here.

File: src/network/modules/blockchain/blockchain.ts [hunk 8]
```diff
@@ -1,17 +1,355 @@
...
+
+  async response(txnHash: ArrayBuffer) {
+    return (await this.call(
+      "blockchain.response",
+      new Map().set(0, new Map().set(0, txnHash)),
+    )) as Message
+  },
+}
+
+async function getBlockainInfo(msg: Message): Promise<InfoReturns> {
```
Comment (stanleyjones) [inline src/network/modules/blockchain/blockchain.ts:189]: typo: Should be `getBlockchainInfo`

**Review by stanleyjones — COMMENTED** [review]
A few comments. I'm especially worried about testing the library from within the apps.

I have to be honest, I don't love how complex the `getBlock`, `getTxn`, and `getRequestOrResponseContent` functions are, but that's sort of the state of things until my declarative branch gets merged. I'll try to get a PR open today.

File: package.json (hunk 3 above)
Comment (mjsorribas) [reply package.json:3]: I agree, also we can add to a document as part of the process of delivering a new release explaining the steps. I'll remove it from this pr.
